src/persist-client/src/cli/inspect.rs

```diff
@@ -599,6 +543,247 @@ pub async fn unreferenced_blobs(
+        "  leaked: {} {} {}",
+        leaked_count,
+        leaked_bytes,
+        human_bytes(leaked_bytes)
+    );
+    println!(
+        "  current writer pending: {} {} {}",
+        current_writer_pending_count,
+        current_writer_pending_bytes,
+        human_bytes(current_writer_pending_bytes)
+    );
+    println!(
+        "  referenced: {} {} {}",
+        referenced_by_state_count,
+        referenced_by_state_bytes,
+        human_bytes(referenced_by_state_bytes)
+    );
+    println!(
+        "    gc-able: {} {} {}",
+        gc_able_parts_count + gc_able_rollups_count,
+        gc_able_parts_bytes + gc_able_rollups_bytes,
+        human_bytes(gc_able_parts_bytes + gc_able_rollups_bytes)
+    );
+    println!(
+        "      gc-able parts: {} {} {}",
+        gc_able_parts_count,
+        gc_able_parts_bytes,
+        human_bytes(gc_able_parts_bytes)
+    );
+    println!(
+        "      gc-able rollups: {} {} {}",
+        gc_able_rollups_count,
+        gc_able_rollups_bytes,
+        human_bytes(gc_able_rollups_bytes)
+    );
+    println!(
+        "    current: {} {} {}",
+        current_parts_count + current_rollups_count,
+        current_parts_bytes + current_rollups_bytes,
+        human_bytes(current_parts_bytes + current_rollups_bytes)
+    );
+    println!(
+        "      current parts: {} {} {}",
+        current_parts_count,
+        current_parts_bytes,
+        human_bytes(current_parts_bytes)
+    );
+    println!(
+        "      current rollups: {} {} {}",
+        current_rollups_count,
+        current_rollups_bytes,
+        human_bytes(current_rollups_bytes)
+    );
+
+    Ok(())
+}
+
+fn human_bytes(bytes: usize) -> String {
+    if bytes < 10_240 {
+        return format!("{}B", bytes);
+    }
+    #[allow(clippy::as_conversions)]
+    let mut bytes = bytes as f64 / 1_024f64;
+    if bytes < 10_240f64 {
+        return format!("{:.1}KiB", bytes);
+    }
+    bytes = bytes / 1_024f64;
+    if bytes < 10_240f64 {
+        return format!("{:.1}MiB", bytes);
+    }
+    bytes = bytes / 1_024f64;
+    if bytes < 10_240f64 {
+        return format!("{:.1}GiB", bytes);
+    }
+    bytes = bytes / 1_024f64;
+    format!("{:.1}TiB", bytes)
+}
+
+impl StateArgs {
+    fn shard_id(&self) -> ShardId {
+        ShardId::from_str(&self.shard_id).expect("invalid shard id")
+    }
+
+    // Intentionally not pub(crate) so we can't accidentally use it from admin.
+    async fn open(&self) -> Result<StateVersions, anyhow::Error> {
+        let cfg = PersistConfig::new(&READ_ALL_BUILD_INFO, SYSTEM_TIME.clone());
+        let metrics = Arc::new(Metrics::new(&cfg, &MetricsRegistry::new()));
+        let consensus = make_consensus(&cfg, &self.consensus_uri, Arc::clone(&metrics)).await?;
+        let blob = make_blob(&self.blob_uri, Arc::clone(&metrics)).await?;
+        Ok(StateVersions::new(cfg, consensus, blob, metrics))
+    }
+}
+
+// Intentionally not pub(crate) so we can't accidentally use it from admin.
```

Comment:
It looks like we could share the implementation with `admin`? Mildly for deduplicating but no strong preference.